src/main/java/com/algorand/algosdk/logic/Logic.java

```diff
@@ -41,17 +41,90 @@
         String[] Group;
     }
 
+    public static class ProgramData {
+        public final boolean good;
+        public final ArrayList<Integer> intBlock;
+        public final ArrayList<byte[]> byteBlock;
+
+        public ProgramData(final boolean good, final ArrayList<Integer> intBlock, final ArrayList<byte[]> byteBlock) {
+            this.good = good;
+            this.intBlock = intBlock;
+            this.byteBlock = byteBlock;
+        }
+    }
+
+    public static class VarintResult {
+        final public int value;
+        final public int length;
+
+        public VarintResult(int value, int length) {
+            this.value = value;
+            this.length = length;
+        }
+
+        public VarintResult() {
+            this.value = 0;
+            this.length = 0;
+        }
+    }
+
+    public static class IntConstBlock {
+        public final int size;
+        public final ArrayList<Integer> results;
+
+        IntConstBlock(final int size, final ArrayList<Integer> results) {
+            this.size = size;
+            this.results = results;
+        }
+    }
+
+    public static class ByteConstBlock {
+        public final int size;
+        public final ArrayList<byte[]> results;
+
+        ByteConstBlock(int size, ArrayList<byte[]> results) {
+            this.size = size;
+            this.results = results;
+        }
+    }
+
+    protected static class Uvarint {
+        public static VarintResult parse(byte[] data) {
+            int x = 0;
+            int s = 0;
+            for (int i = 0; i < data.length; i++) {
+                int b = data[i] & 0xff;
+                if (b < 0x80) {
+                    if (i > 9 || i == 9 && b > 1) {
+                        return new VarintResult(0, -(i + 1));
+                    }
+                    return new VarintResult(x | (b & 0xff) << s, i + 1);
+                }
+                x |= ((b & 0x7f) & 0xff) << s;
+                s += 7;
+            }
+            return new VarintResult();
+        }
+    }
+
     private static LangSpec langSpec;
     private static Operation[] opcodes;
 
+    public static boolean checkProgram(byte[] program, ArrayList<byte[]> args) throws IOException {
```

Comment:
Since checkProgram is only used in the test, it is better if it is a test helper method. 
It is not needed here.

Reply:
This method isn't actually new, I must have moved it from elsewhere in the code. It's used by the LogicSignature class.

Reply:
checkProgram is used on LogicSig creation when a program supplied by a user, so it is not test-only helper